mapit_labour/management/commands/mapit_labour_import_addressbase_core.py

```diff
@@ -0,0 +1,157 @@
+    label = "<AddressBase Core CSV file>"
+
+    count = {}  # initialised in handle()
+    batch_size = 1000
+    purge = False
+    limit = 0
+
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument(
+            "--purge",
+            action="store_true",
+            dest="purge",
+            default=self.purge,
+            help="Purge all existing UPRNs and import afresh",
+        )
+        parser.add_argument(
+            "--batch-size",
+            dest="batch_size",
+            type=int,
+            default=self.batch_size,
+            help=f"Batch size for bulk INSERT/UPDATE operations. Default {self.batch_size}",
+        )
+        parser.add_argument(
+            "--limit",
+            dest="limit",
+            type=int,
+            default=self.limit,
+            help="Stop after importing this many rows from the CSV. 0 (default) for no limit.",
+        )
+
+    def handle_label(self, label: str, **options):
+        self.purge = options["purge"]
+        self.batch_size = options["batch_size"]
+        self.limit = options["limit"]
+
+        with open(label, encoding="utf-8-sig") as f:
+            self.handle_rows(DictReader(f))
+
+    def handle(self, *args, **kwargs):
+        self.count = {
+            "total": 0,
+            "created": 0,
+            "updated": 0,
+        }
+        super().handle(*args, **kwargs)
+
+    def handle_rows(self, csv: DictReader):
+        if self.purge:
+            UPRN.objects.all().delete()
+
+        for rows in batched(islice(csv, self.limit or None), self.batch_size):
+            if self.purge:
+                new, existing = rows, []
+            else:
+                new, existing = self.find_existing_uprns(rows)
+            if new:
+                with transaction.atomic():
+                    UPRN.objects.bulk_create(self.create_uprn(row) for row in new)
+            if existing:
+                self.update_existing_uprns(existing)
+            self.print_stats()
+        self.print_stats()
+
+    def create_uprn(self, row: Dict[str, str]):
+        row = {k.lower(): v for k, v in row.items()}
+
+        self.count["total"] += 1
+        self.count["created"] += 1
+        return UPRN(
+            uprn=row["uprn"],
+            postcode=row["postcode"].replace(" ", ""),
+            location=Point(float(row["easting"]), float(row["northing"]), srid=27700),
+            single_line_address=row["single_line_address"],
+            addressbase=row,
+        )
+
+    def find_existing_uprns(self, rows):
+        """
+        Takes a list of rows from the CSV and divides them into two lists which
+        are returned: those that don't already exist as URPN objects in the DB
+        and those that do.
+        """
+        # need to consume this iterator now because we use it twice below
+        rows = list(rows)
+        db_uprns = {u.uprn: u for u in UPRN.objects.filter(uprn__in=(r["UPRN"] for r in rows))}
```

Comment:
This change looks good to me, makes it a bit simpler doesn't it. I used `black` on this project so please format this line as so:

```suggestion
        db_uprns = {
            u.uprn: u for u in UPRN.objects.filter(uprn__in=(r["UPRN"] for r in rows))
        }
```